### PR TITLE
Resource dump authentication support

### PR DESCRIPTION
**Pull request by Pavithrab7**
Upstream commits - 
https://gerrit.openbmc-project.xyz/c/openbmc/pldm/+/45156/7
https://gerrit.openbmc-project.xyz/c/openbmc/pldm/+/49166/5
https://gerrit.openbmc-project.xyz/c/openbmc/pldm/+/49211/3
https://gerrit.openbmc-project.xyz/c/openbmc/pldm/+/49404/3

### CODE REVIEW

File: oem/ibm/libpldmresponder/file_io.cpp [hunk 1]
```diff
@@ -930,13 +930,61 @@ Response Handler::fileAckWithMetaData(const pldm_msg* request,
...
     auto responsePtr = reinterpret_cast<pldm_msg*>(response.data());
     encode_file_ack_with_meta_data_resp(request->hdr.instance_id, rc,
                                         responsePtr);
     return response;
 }
 
+Response Handler::newFileAvailableWithMetaData(const pldm_msg* request,
+                                               size_t payloadLength)
+{
+    std::cerr << " Got a newFile available with metadata command" << std::endl;
```
Comment (jaypadath) [inline oem/ibm/libpldmresponder/file_io.cpp:944]: pls remove this also ? or if you want to keep make it as std::cout as its not an error

File: oem/ibm/libpldmresponder/file_io_type_dump.cpp [hunk 2]
```diff
@@ -287,5 +287,130 @@ int DumpHandler::read(uint32_t offset, uint32_t& length, Response& response,
...
+int DumpHandler::fileAckWithMetaData(uint32_t metaDataValue1,
+                                     uint32_t metaDataValue2,
+                                     uint32_t /*metaDataValue3*/,
+                                     uint32_t /*metaDataValue4*/)
+{
+    auto path = findDumpObjPath(fileHandle);
+    uint8_t fileStatus = (uint8_t)metaDataValue2;
+    if (dumpType == PLDM_FILE_TYPE_RESOURCE_DUMP_PARMS)
+    {
+        std::cerr << "Token: " << metaDataValue1 << std::endl;
```
Comment (jaypadath) [inline oem/ibm/libpldmresponder/file_io_type_dump.cpp:299]: if you want to keep can u pls make it as std::cout as its not an error

File: oem/ibm/libpldmresponder/file_io_type_dump.cpp [hunk 3]
```diff
@@ -287,5 +287,130 @@ int DumpHandler::read(uint32_t offset, uint32_t& length, Response& response,
...
+
+    return PLDM_ERROR;
+}
+
+int DumpHandler::newFileAvailableWithMetaData(uint64_t length, uint32_t token)
+{
+    static constexpr auto dumpInterface = "xyz.openbmc_project.Dump.NewDump";
+    auto& bus = pldm::utils::DBusHandler::getBus();
+
+    std::cerr << "Token sent:" << token << std::endl;
```
Comment (jaypadath) [inline oem/ibm/libpldmresponder/file_io_type_dump.cpp:388]: pls make to std::cout

File: oem/ibm/libpldmresponder/file_io_type_dump.cpp (hunk 2 above)
Comment (Pavithrab7) [reply oem/ibm/libpldmresponder/file_io_type_dump.cpp]: done

File: oem/ibm/libpldmresponder/file_io_type_dump.cpp (hunk 3 above)
Comment (Pavithrab7) [reply oem/ibm/libpldmresponder/file_io_type_dump.cpp]: done

**Review by jaypadath — APPROVED** [review]

**Pull request merged**